sed/core/processor.py

```diff
@@ -1702,6 +1702,7 @@ def align_dld_sectors(
     def calibrate_delay_axis(
         self,
         delay_range: Tuple[float, float] = None,
+        read_delay_ranges: bool = True,
```

Comment:
If this flag is set to true by default, wouldn't it lead to an error for cases where data file is not available to read? Such as in the case of Flash data

Reply:
This case raises indeed an error. However, this function is anyways not intended for use with Flash data. It converts an ADC signal into a delay signal, and you directly use the delay stage values at Flash, no?
For using the stored calibration data regardless of the values in data files, I added this option, mostly for consistency with the config format. Honestly, I don't see a large use case for this option to load delay calibration data anyways, because typically they are different for each run, so any saved config is not relevant for a different scan.

Reply:
You're right about Flash data but this could pose a problem for future loaders utilizing ADC signal. If you find that this isn't a big issue, everything seems fine to me.

> For using the stored calibration data regardless of the values in data files, I added this option, mostly for consistency with the config format. Honestly, I don't see a large use case for this option to load delay calibration data anyways, because typically they are different for each run, so any saved config is not relevant for a different scan.

Does the user create different objects or calibration files for each run? Or is this an automated check based on metadata (such as runnumber)?

Reply:
> Does the user create different objects or calibration files for each run? Or is this an automated check based on metadata (such as runnumber)?

Normally, the delay range is automatically read from each file, and applied with this function, without the need for saving the calibration dict. This uses the default adc_ranges set in the config. Just reconfiguring the adc_ranges and still taking delay_ranges from the file did not work if exectuted twice in the current main branch. This PR fixes this logic, so that this works as default. If one wants to instead use the saved config, the new flag has to be saved.